span.go

```diff
@@ -47,6 +35,46 @@ type Span interface {
 	// Suitable for serializing over the wire, etc.
 	TraceContext() TraceContext
 
+	// Log() records `data` to this Span at `timestamp`.
+	//
+	// See LogData for semantic details.
+	Log(timestamp time.Time, data LogData)
+
+	// Info() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprint(args...)})
+	//
+	Info(args ...interface{})
+
+	// Infof() is equivalent to
+	//
+	//   Log(time.Now(), LogData{Message: fmt.Sprintf(format, args...)})
```

Comment:
PS: I am not eager to sort out lazy-eval without reliable profiling data from multiple systems/deployments since it's basically an optimization.

PPS: Not sure it's wise to include both `Info` and `Infof` (etc)... perhaps we should just choose an idiom and stick to it.